cranelift/codegen/src/ir/dfg.rs

```diff
@@ -226,8 +230,12 @@ impl DataFlowGraph {
     }
 
     /// Make a BlockCall, bundling together the block and its arguments.
-    pub fn block_call(&mut self, block: Block, args: &[Value]) -> BlockCall {
-        BlockCall::new(block, args, &mut self.value_lists)
+    pub fn block_call<'a>(
+        &mut self,
+        block: Block,
+        args: impl IntoIterator<Item = &'a BlockArg>,
```

Comment:
It would have been nice if this accepted `impl IntoIterator<Item = BlockArg>`. `BlockArg` is `Copy`, so you can use `.copied()` on the iterator if necessary to convert from an `&BlockArg` to a `BlockArg` iterator. However if you are trying to build the `BlockArgs`s on the fly, you are forced to collect them into a `Vec` before passing it to `block_call` with the current function signature.

Reply:
IIRC, this was necessary to allow `&[a, b]`-style arg-lists, which are everywhere in other crates in-repo, to continue to work. Although now that I think about it, I suppose just `[a, b]` might work with `Item = BlockArg`. I don't have strong feelings here and I'm happy to review a PR if you can find a way to make it nicer (including with existing callsites)...